src/mgmt/cli_mgmt.c

```diff
@@ -849,43 +852,72 @@ dc_mgmt_get_pool_svc_ranks(struct dc_mgmt_sys *sys, const uuid_t puuid,
 	struct mgmt_pool_get_svcranks_in       *rpc_in;
 	struct mgmt_pool_get_svcranks_out      *rpc_out;
 	crt_opcode_t				opc;
-	int					rc;
+	int					i;
+	int					idx;
+	bool					success = false;
+	int					rc = 0;
+
+	/* NB: sy_psrs[] may have multiple entries even for single MS replica,
+	 * since there may be multiple ioservers there. Some of which may have
+	 * been stopped or faulted. May need to contact multiple ioservers.
+	 * Assumed: any MS replica ioserver can be contacted, even non-leaders.
+	 */
+	D_ASSERT(sys->sy_npsrs > 0);
+	idx = rand() % sys->sy_npsrs;
+	for (i = 0 ; i < sys->sy_npsrs; i++) {
+		srv_ep.ep_grp = sys->sy_group;
+		srv_ep.ep_rank = sys->sy_psrs[idx].rank;
+		srv_ep.ep_tag = daos_rpc_tag(DAOS_REQ_MGMT, 0);
+
+		rpc = NULL;
+		opc = DAOS_RPC_OPCODE(MGMT_POOL_GET_SVCRANKS, DAOS_MGMT_MODULE,
+				      DAOS_MGMT_VERSION);
+		rc = crt_req_create(daos_get_crt_ctx(), &srv_ep, opc, &rpc);
```

Comment:
do we need to call daos_get_crt_ctx() on every iteration?

Reply:
I guess not. I will pull it out of the loop and likely remove some other statements that only need to be done once.